llama-index-integrations/embeddings/llama-index-embeddings-oracleai/llama_index/embeddings/oracleai/base.py

```diff
@@ -0,0 +1,167 @@
+                    SYS.DBMS_VECTOR.load_onnx_model(:path, :filename, :model, json('{"function" : "embedding", "embeddingOutput" : "embedding" , "input": {"input": ["DATA"]}}'));
+                end;""",
+                path=dir,
+                filename=onnx_file,
+                model=model_name,
+            )
+
+            cursor.close()
+
+        except Exception as ex:
+            print(f"An exception occurred :: {ex}")
+            cursor.close()
+            raise
+
+    def _get_embedding(self, text: str) -> List[float]:
+        try:
+            import oracledb
+        except ImportError as e:
+            raise ImportError(
+                "Unable to import oracledb, please install with "
+                "`pip install -U oracledb`."
+            ) from e
+
+        if text is None:
+            return None
+
+        embedding = None
+        try:
+            oracledb.defaults.fetch_lobs = False
+            cursor = self._conn.cursor()
+
+            if self._proxy:
+                cursor.execute(
+                    "begin utl_http.set_proxy(:proxy); end;", proxy=self._proxy
+                )
+
+            cursor.execute(
+                "select t.* from dbms_vector_chain.utl_to_embeddings(:content, json(:params)) t",
+                content=text,
+                params=json.dumps(self._params),
+            )
+
+            row = cursor.fetchone()
+            if row is None:
+                embedding = []
+            else:
+                rdata = json.loads(row[0])
+                # dereference string as array
+                embedding = json.loads(rdata["embed_vector"])
+
+            cursor.close()
+            return embedding
+        except Exception as ex:
+            print(f"An exception occurred :: {ex}")
+            cursor.close()
+            raise
+
+    def _get_text_embedding(self, text: str) -> List[float]:
+        return self._get_embedding(text)
+
+    def _get_query_embedding(self, query: str) -> List[float]:
+        return self._get_embedding(query)
+
+    async def _aget_query_embedding(self, query: str) -> List[float]:
+        return self._get_query_embedding(query)
+
+
+# uncomment the following code block to run the test
+
+"""
+# A sample unit test.
+
+''' get the Oracle connection '''
+conn = oracledb.connect(
+    user="<username>",
+    password="<password>",
+    dsn="<hostname/service_name>")
+print("Oracle connection is established...")
+
+''' params '''
+embedder_params = {"provider":"database", "model":"demo_model"}
+proxy = "<proxy>"
+
+''' instance '''
+embedder = OracleEmbeddings(conn=conn, params=embedder_params, proxy=proxy)
+
+embed = embedder._get_text_embedding("Hello World, Text!")
+print(f"Embedding generated by OracleEmbeddings: {embed}")
+
+embed = embedder._get_query_embedding("Hello World, Query!")
+print(f"Embedding generated by OracleEmbeddings: {embed}")
+
+conn.close()
+print("Connection is closed.")
+
+"""
```

Comment:
Let's remove this. We can put tests in an actual test file (and mark them as skip if they rely on env variables that are missing)

Reply:
Each file has a quick unit test (being commented out) just to verify the code working fine after each update. Our QA team will keep adding more tests in the test directory. I thought it's good to have a quick unit test for a quick verification. However, I can remove if needed. Let me know.

Reply:
Hari it is okay to remove

Reply:
Yea, I would prefer unit tests to go into the `tests/` folder, just to match the coding style of the codebase

Reply:
removed.